src/System.Net.Security/src/System/Net/SecureProtocols/SslState.cs

```diff
@@ -705,10 +705,7 @@ internal void EndProcessAuthentication(IAsyncResult result)
                 throw new ArgumentException(SR.Format(SR.net_io_async_result, result.GetType().FullName), "asyncResult");
             }
 
-            if (Interlocked.Exchange(ref _nestedAuth, 0) == 0)
-            {
-                throw new InvalidOperationException(SR.Format(SR.net_io_invalidendcall, "EndAuthenticate"));
-            }
+            _nestedAuth = 0;
```

Comment:
I think we should remove only the check but still use Interlocked.Exchange in case another authentication request is started.